omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResource.java

```diff
@@ -0,0 +1,311 @@
+            }
+            if (StringUtils.isNotBlank(batchSize)) {
+                if(StringUtils.isNumeric(batchSize)) {
+                    identifierSource.setBatchSize(Integer.parseInt(batchSize));
+                }else {
+                    errors.add("batch size");
+                }
+            }
+            if (StringUtils.isNotBlank(minPoolSize)) {
+                if(StringUtils.isNumeric(minPoolSize)) {
+                    identifierSource.setMinPoolSize(Integer.parseInt(minPoolSize));
+                }else {
+                    errors.add("min pool size");
+                }
+            }
+            if (StringUtils.isNotBlank(sequential)) {
+                if(validateBoolean(sequential) != null) {
+                    identifierSource.setSequential(validateBoolean(sequential));
+                }else {
+                    errors.add("sequential");
+                }
+            }
+            if (StringUtils.isNotBlank(refillWithScheduledTask)) {
+                if(validateBoolean(refillWithScheduledTask) != null) {
+                    identifierSource.setSequential(validateBoolean(refillWithScheduledTask));
+                }else {
+                    errors.add("refill");
+                }
+            }
+            if(errors.toString().length() > 2) {
+                throw new ConversionException("The values of the following inputs are missing or invalid: " + errors.toString());
+            }
+
+            identifierSource.setIdentifierType(patientIdentifierType);
+            identifierSource.setName(name);
+            identifierSource.setDescription(description);
+
+            savedIdentifierSource =  save(identifierSource);
+        }
+        return ConversionUtil.convertToRepresentation(savedIdentifierSource, Representation.DEFAULT);
+    }
+	
+    @Override
+    public IdentifierSource save(IdentifierSource identifierSource) {
+        return Context.getService(IdentifierSourceService.class).saveIdentifierSource(identifierSource);
+    }
+
+    @Override
+    public IdentifierSource getByUniqueId(String uniqueId) {
+        return Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(uniqueId);
+    }
+	
+    @Override
+    protected void delete(IdentifierSource identifierSource, String reason, RequestContext context) throws ResponseException {
+        Context.getService(IdentifierSourceService.class).retireIdentifierSource(identifierSource, reason);
+    }
+	
+    @Override
+    public void purge(IdentifierSource identifierSource, RequestContext context) throws ResponseException {
+        Context.getService(IdentifierSourceService.class).purgeIdentifierSource(identifierSource);
+    }
+
+    private String validateInput(Object input) {
+        if (input == null || input.toString().length() == 0) {
+                return "";
+        }
+        return input.toString();
+    }
+    
+    private Boolean validateBoolean(String value) {
```

Comment:
Is this validateBoolean or parseBoolean?

Reply:
fixed